pysat/_instrument.py

```diff
@@ -2048,363 +1584,909 @@ def bounds(self, value=None):
-            for inst in inst:
-                print('Another day loaded', inst.date)
+    @date.setter
+    def date(self, new):
+        """Date for loaded data."""
+        self._date = self._filter_datetime_input(new)
+
+    @property
+    def index(self):
+        """Returns time index of loaded data."""
+        return self._index()
+
+    @property
+    def variables(self):
+        """Returns list of variables within loaded data."""
+
+        if self.pandas_format:
+            return self.data.columns
+        else:
+            return list(self.data.variables.keys())
+
+    def copy(self):
+        """Deep copy of the entire Instrument object."""
+
+        return copy.deepcopy(self)
+
+    def concat_data(self, new_data, prepend=False, **kwargs):
+        """Concats new_data to self.data for xarray or pandas as needed
+
+        Parameters
+        ----------
+        new_data : pds.DataFrame, xr.Dataset, or list of such objects
+            New data objects to be concatonated
+        prepend : boolean
+            If True, assign new data before existing data; if False append new
+            data (default=False)
+        **kwargs : dict
+            Optional keyword arguments passed to pds.concat or xr.concat
+
+        Note
+        ----
+        For pandas, sort=False is passed along to the underlying
+        pandas.concat method. If sort is supplied as a keyword, the
+        user provided value is used instead.  Recall that sort orders the
+        data columns, not the data values or the index.
+
+        For xarray, dim=Instrument.index.name is passed along to xarray.concat
+        except if the user includes a value for dim as a keyword argument.
 
         """
+        # Order the data to be concatonated in a list
+        if not isinstance(new_data, list):
+            new_data = [new_data]
 
-        if self._iter_type == 'file':
-            for fname in self._iter_list:
-                self.load(fname=fname)
-                yield self
+        if prepend:
+            new_data.append(self.data)
+        else:
+            new_data.insert(0, self.data)
 
-        elif self._iter_type == 'date':
-            for date in self._iter_list:
-                self.load(date=date)
-                yield self
+        # Retrieve the appropriate concatonation function
+        if self.pandas_format:
+            # Specifically do not sort unless otherwise specified
+            if 'sort' not in kwargs:
+                kwargs['sort'] = False
+            concat_func = pds.concat
+        else:
+            # Specify the dimension, if not otherwise specified
+            if 'dim' not in kwargs:
+                kwargs['dim'] = self.index.name
+            concat_func = xr.concat
+
+        # Assign the concatonated data to the instrument
+        self.data = concat_func(new_data, **kwargs)
+        return
+
+    def today(self):
+        """Returns today's date, with no hour, minute, second, etc.
```

Comment:
I don't think it's necessary because we consistently use date to refer to UTC date.  However, I am not opposed to greater clarity.  Do you want to alter that as a suggestion?  Then I can batch commit it with the typo you found.